src/Microsoft.ML.Core/Data/Repository.cs

```diff
@@ -333,10 +334,8 @@ public Entry CreateEntry(string dir, string name)
             string pathEnt;
             string pathTemp;
             GetPath(out pathEnt, out pathTemp, dir, name, true);
-            if (PathMap.ContainsKey(pathEnt))
+            if (PathMap.TryAdd(pathEnt, pathTemp))
```

Comment:
Assuming I'm reading the docs right, `TryAdd()` returns `false` when it doesn't add ([docs](https://docs.microsoft.com/en-us/dotnet/api/system.collections.concurrent.concurrentdictionary-2.tryadd?view=netcore-3.0#examples)). 

Hence need to invert the return value.
```suggestion
            if (!PathMap.TryAdd(pathEnt, pathTemp))
```

Reply:
Whoops! I thought I added that, but I must have missed it. Thanks for catching!